src/acceptance-test/java/org/zalando/nakadi/webservice/EventStreamReadingAT.java

```diff
@@ -301,6 +302,68 @@ public void whenInvalidCursorsThenPreconditionFailed() {
                 .body("detail", equalTo("non existing partition very_wrong_partition"));
     }
 
+    @Test(timeout = 10000)
+    public void whenReadEventsForBlockedConsumerThen429() throws Exception {
+        final FloodService.Flooder flooder = new FloodService.Flooder(EVENT_TYPE_NAME, FloodService.Type.CONSUMER_ET);
+
+        given()
+                .header(new Header("X-nakadi-cursors", xNakadiCursors))
+                .param("batch_limit", "5")
+                .param("stream_timeout", "2")
+                .param("batch_flush_timeout", "2")
+                .when()
+                .get(STREAM_ENDPOINT)
+                .then()
+                .statusCode(HttpStatus.OK.value());
+
+        NakadiControllerAT.blockFlooder(flooder);
+        given()
+                .header(new Header("X-nakadi-cursors", xNakadiCursors))
+                .param("batch_limit", "5")
+                .param("stream_timeout", "2")
+                .param("batch_flush_timeout", "2")
+                .when()
+                .get(STREAM_ENDPOINT)
+                .then()
+                .statusCode(HttpStatus.TOO_MANY_REQUESTS.value())
+                .header("Retry-After", "300");
+
+        NakadiControllerAT.unblockFlooder(flooder);
+        given()
+                .header(new Header("X-nakadi-cursors", xNakadiCursors))
+                .param("batch_limit", "5")
+                .param("stream_timeout", "2")
+                .param("batch_flush_timeout", "2")
+                .when()
+                .get(STREAM_ENDPOINT)
+                .then()
+                .statusCode(HttpStatus.OK.value());
+    }
+
+    @Test(timeout = 10000)
+    public void whenReadEventsConsumerIsBlocked() throws Exception {
```

Comment:
@v-stepanov actually, i test that the consumer was blocked during consumption but to test that I use test timeout, right.